18F/identity-idp
# Simplify session with trust check to only consider user

## 🛠 Summary of changes

Replaces `session_with_trust?` with `user_signed_in?`, removing consideration of current page.

`Users::SessionController#new` cannot be visited if a user is present ([source](https://github.com/18F/identity-idp/blob/3ffdd801d652099fdbf4275406b60845e31475fe/app/controllers/users/sessions_controller.rb#L93)), so the second part of this condition is redundant.

This also fixes an issue obfuscated by previous tests where the return value was not always a boolean but could instead return the full user object when present. It now always returns a boolean.

## 📜 Testing Plan

Verify DAP is loaded only if signed out.

1. Set `participate_in_dap: true` in `config/application.yml`
2. Visit http://localhost:3000
3. Inspect page to verify DAP is present
4. Click "Forgot your password?"
5. Inspect page to verify DAP is present
6. Return to http://localhost:3000
7. Click "Create an account"
8. Inspect page to verify DAP is present
9. Return to http://localhost:3000
10. Sign in
11. Inspect page to verify DAP is NOT present
